data-products/src/test/scala/org/sunbird/analytics/sourcing/TestSourcingMetrics.scala

```diff
@@ -0,0 +1,84 @@
+    val encoders = Encoders.product[ContentHierarchy]
+    val data = spark.read.format("org.apache.spark.sql.cassandra").options(Map("table" -> "content_hierarchy", "keyspace" -> sunbirdHierarchyStore)).load()
+      .as[ContentHierarchy](encoders).first()
+    val hierarchy = JSONUtils.deserialize[TextbookHierarchy](data.hierarchy)
+    val report = SourcingMetrics.generateReport(List(hierarchy),List(), List(),hierarchy,List(),List("","0"))
+    val df = report._1.toDF().withColumn("slug",lit("test-slug")).withColumn("reportName",lit("TextbookReport"))
+    val configMap = JSONUtils.deserialize[Map[String,AnyRef]](config)
+    val reportConfig = configMap("modelParams").asInstanceOf[Map[String, AnyRef]]("reportConfig").asInstanceOf[Map[String, AnyRef]]
+    SourcingMetrics.saveReportToBlob(df,JSONUtils.serialize(reportConfig),StorageConfig("local","test","Textbook",Option(""),Option("")),"TextbookReport", "sourcing")
+
+    report._1.length should be (1)
+    report._2.length should be (2)
+    report._2.map(f => f.contentType should not be("TextbookUnit"))
```

Comment:
Not enough assertions. Need to assert for each column and the variations of column data

Reply:
Added assertions for all columns